pydantic_core/_types.py

```diff
@@ -0,0 +1,162 @@
+from __future__ import annotations
+
+import sys
+from typing import Any, Callable, Dict, List, Sequence, Union
+
+if sys.version_info < (3, 11):
+    from typing_extensions import NotRequired
+else:
+    from typing import NotRequired
+
+if sys.version_info < (3, 8):
+    from typing_extensions import Literal, TypedDict
+else:
+    from typing import Literal, TypedDict
+
+
+class AnySchema(TypedDict):
+    type: Literal['any']
+
+
+class BoolSchema(TypedDict):
+    type: Literal['bool']
+    strict: NotRequired[bool]
+
+
+class ConfigSchema(TypedDict):
+    strict: NotRequired[bool]
+    extra: NotRequired[Literal['allow', 'forbid', 'ignore']]
+
+
+class DictSchema(TypedDict):
+    type: Literal['dict']
+    keys: Schema  # type: ignore[misc]
+    values: Schema  # type: ignore[misc]
+    min_items: NotRequired[int]
+    max_items: NotRequired[int]
+
+
+class FloatSchema(TypedDict):
+    type: Literal['float']
+    multiple_of: NotRequired[float]
+    le: NotRequired[float]
+    ge: NotRequired[float]
+    lt: NotRequired[float]
+    gt: NotRequired[float]
+    strict: NotRequired[bool]
+    default: NotRequired[float]
+
+
+# TODO: function could be typed based on mode
+class FunctionSchema(TypedDict):
+    type: Literal['function']
+    mode: Literal['before', 'after', 'plain', 'wrap']
+    function: Callable[..., Any]
+    schema: NotRequired[Schema]  # type: ignore[misc]
+
+
+class IntSchema(TypedDict):
+    type: Literal['int']
+    multiple_of: NotRequired[int]
+    le: NotRequired[int]
+    ge: NotRequired[int]
+    lt: NotRequired[int]
+    gt: NotRequired[int]
+    strict: NotRequired[bool]
+
+
+class ListSchema(TypedDict):
+    type: Literal['list']
+    items: Schema  # type: ignore[misc]
```

Comment:
It seems from source that it will allow Any type, but unlike dict it doesn't actually build an `Any` validator. Should we add an `Any` default? Might also want to match up the implementations in dict and list, even if it is an implementation detail.

Reply:
You can't add defaults on a typed dict so I just left a comment `# default: AnySchema`.